### DISCUSSION TITLE
ChampSim Benchmarking Workloads

### DISCUSSION THREAD

**Discussion by yardboy27** [ChampSim/ChampSim#513] [Q&A] [answered]
Are there any specific trace workloads that are used as benchmarks for ChampSim releases? Is there a standard benchmark workload that is run with every stable ChampSim release with the default ChampSim configuration, and if there is, where that benchmark workload data is published?

I was asked recently by a project mentor how I knew that ChampSim was running a specific workload correctly, and they asked if I had run a general benchmarking workload on ChampSim to verify that everything was working correctly. I did not have a good answer for them, so I thought I would reach out to the community and ask them how they know ChampSim is working correctly. Thank you in advance!

**Comment by ngober** [comment, marked as answer]
I hear questions like this somewhat often, and they're always difficult to answer because the word "correct" is not well-defined in this context. What is the correct way to run a benchmark workload? How would you know that it is running "correctly" versus "incorrectly"? The answers are, surprisingly, quite up to interpretation.

That said, there are ways that we can verify that ChampSim is doing what it claims to do. ChampSim does this through its software tests. We can [verify](https://github.com/ChampSim/ChampSim/blob/aa336e838027563196eeb721be027f48d63a1b39/test/cpp/src/401-hit-latency.cc#L7) that the cache model demonstrates hits on repeated requests, that the observed latency is equal to the configured hit latency, and that the cache reports a hit to the statistics. We can [verify](https://github.com/ChampSim/ChampSim/blob/aa336e838027563196eeb721be027f48d63a1b39/test/cpp/src/431-merge-prefetch.cc#L59) that a prefetch that matches an inflight MSHR gets dropped, or [that](https://github.com/ChampSim/ChampSim/blob/aa336e838027563196eeb721be027f48d63a1b39/test/cpp/src/431-merge-prefetch.cc#L83) a late prefetch gets promoted to a demand. If we have confidence in the parts and confidence in the integration, we can have confidence in the results.
